src/Bicep.Core/Syntax/Converters/ProgramSyntaxConverter.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Core.Syntax.Converters
+{
+    public static class ProgramSyntaxConverter
```

Comment:
>ProgramSyntaxConverter [](http://example.com/codeflow?start=24&length=22)

Now that we inverted the dependencies in the Template library so it doesn't pull in the schemas, any value in using the library in the converter instead of parsing the template ourselves?

Reply:
That's a good idea and I'd like to switch to the Template library as we will be able to validate the whole template rather than just the parameter and output type definitions. However, currently we are not using the `Azure.Deployments.Templates` package which contains the template engine needed to do template validation. Any objections if I add a reference to the package to bicep core?

Reply:
After checking the source code I feel like there's no harm in using `Azure.Deployments.Templates`, so I updated the converters to use the library, and I was able to remove most of the custom template validation logics.